org.lflang/src/org/lflang/generator/c/CGenerator.xtend

```diff
@@ -873,29 +837,44 @@ class CGenerator extends GeneratorBase {
         refreshProject()
     }
     
+    
     /**
-     * Generate preambles for 'federate' provided that the preambles in question
-     * are not already in the .lf file that contains the main reactor. 
-     * Therefore, either there is no main reactor or the federate is imported from
-     * another .lf file.
+     * Look at the 'reactor' eResource.
+     * If it is an imported .lf file, incorporate it into the current 
+     * program in the following manner:
+     * - If there are any cmake-include files, add them to the current list
+     *  of cmake-include files.
+     * - If there are any preambles, add them to the preambles of the reactor.
      */
-    def generatePreambleForFederate(FederateInstance federate) {
-        // Check if it is imported. If not imported, the preamble is already contained in the
-        // main resource and has already been added.
-        var Resource mainResource = null;
-        if (mainDef !== null) {
-            mainResource = mainDef.reactorClass.eResource
-        }
-        if (federate.instantiation !== null &&
-            federate.instantiation.reactorClass.eResource !== null &&
-            federate.instantiation.reactorClass.eResource != mainResource
+    def inspectReactorEResource(ReactorDecl reactor) {
+        // If the reactor is imported, look at the
+        // target definition of the .lf file in which the reactor is imported from and
+        // append any cmake-include.
+        // Check if the reactor definition is imported
+        if (reactor.toDefinition.eResource !== 
+            mainDef.reactorClass.toDefinition.eResource
         ) {
             // Extract the contents of the imported file
-            val contents = federate.instantiation.reactorClass.toDefinition.eResource.contents;
+            val contents = reactor.toDefinition.eResource.contents;
             val model = contents.get(0) as Model
-            for (p : model.preambles?: emptyList) {
-                pr(p.code.toText)
+            for (c : model.eContents) {
+                if (c instanceof TargetDecl) {
```

Comment:
There will likely be other target properties for which we want to do something similar. Rather than replicate logic for finding target properties and manually updating a `TargetConfig` instance, I think we should factor out and reuse the `setTargetConfig` method that's currently in `GeneratorBase`. We can then just create a separate `TargetConfig` for the imported file and merge its contents with the main instance as deemed fit. I think it makes sense to either turn `setTargetConfig` into a constructor for `TargetConfig` or make it a factory method.

Reply:
I agree that this is a good idea. However, there will be issues regarding the merging policy of target properties and issues related to federates (e.g., `cmake-include` might need to be federate-based). This looks to be out of scope of this PR. Would you like to create an issue for this?

Reply:
Not sure I agree that it is out-of-scope, because this PR already introduces code replication through the way this is handled. Do you want me to take a stab at this and push a fix to this branch?

Reply:
I'm not sure what code replication you are referring to. This piece of code is just moved  from `doGenerate` to a separate `inspectReactorEResource` function.

Reply:
I'm referring to the fact that the code below replicates what the `TargetProperty`/`TargetConfig` classes were already designed to do.

Reply:
Well, this is a piece of code that was already there. I think if we want to address this issue, a branch named `target-config-import` or something to that effect would be more appropriate than just a `cmake-include-improvements`. The former has a much broader implication on all targets and needs quite a bit of thought and discussions. The latter is much more limited in scope and only applies mostly to the `cmake-include` property. The changes are even further only limited to the C target.

Reply:
The change I suggested here is also strictly limited to the `cmake-include` property and doesn't require much though or discussion IMO, but if you prefer I address it in a separate branch so you go ahead and merge this, I'm also fine with that.

Reply:
I can take a stab at fixing it for `cmake-include`.

Reply:
This issue I believe has been fixed 5ee652f.

Reply:
Doesn't look like this commit exists anymore (the link is broken).